00_setup.Rmd

```diff
@@ -61,6 +83,316 @@ minaantal_tub_group <- as.integer(tot_n_tub/aantal_strat)
+  output_vc <- write_vc(gw_types, file.path(".","data","gw_types"), sorting = c("type"), strict =  FALSE)
+  
+  types <- read_types(lang = "nl")
+  output_vc <- write_vc(types, file.path(".","data","types"), sorting = c("type"), strict =  FALSE)
+  
+  
+  rm(output_vc)
+}
+```
+
+```{r read-gisdata, eval= TRUE, include= TRUE, echo= FALSE}
+if (refresh_data == 2) {
+  habmap_terr <- read_habitatmap_terr()
+  
+  habmap_polygons <- habmap_terr$habitatmap_terr_polygons
+  
+  habmap_types <- habmap_terr$habitatmap_terr_types
+  
+  habmap_types <- habmap_types %>%
+    mutate( polygon_id = as.factor(.data$polygon_id),
+            certain = .data$certain == 1,
+            type = factor(.data$type,
+                          levels = levels(types$type)
+            )
+    )
+  
+  # Verspreiding van de verdrogingsgevoelige typen in Vlaanderen volgens de habitatkaart.
+  habmap_types_gw <- habmap_types %>% 
+    inner_join(gw_types, by = c("type" = "type"))
+  
+  habmap_polygons_gw <- 
+    habmap_polygons %>% 
+    inner_join(habmap_types_gw %>% 
+                 dplyr::select(-code_orig), 
+               by = "polygon_id")
+  
+  #wegschrijven als geopackage
+  datapath <- "G:/Mijn Drive/PRJ_Meetnet_Droogte/2_Uitvoering/data"
+  dir.create(file.path(datapath, "GIS/VoorR/"), recursive = TRUE)
+  
+  st_write(habmap_polygons_gw,
+           file.path(datapath, 
+                     "GIS/VoorR/habmap_terr_gw.gpkg"), 
+           layer = "habitatmap_terr_polygons_gw", 
+           driver = "GPKG",
+           delete_dsn = TRUE)
+  
+  con = dbConnect(SQLite(),
+                  dbname = file.path(
+                    datapath, 
+                    "GIS/VoorR/habmap_terr_gw.gpkg")
+                  )
+  
+  dbWriteTable(con, "habitatmap_terr_types_gw", habmap_types_gw)
+  dbDisconnect(con)
+  
+  
+  #GRTS-data
+  
+  raster_meetnet_poly <- read_GRTSmh_diffres(level = 8, polygon = TRUE)
+  raster_meetnet_poly <- raster_meetnet_poly %>% 
+    rename(rasterid =  value)
+  
+  #wegschrijven van dit grid
+  st_write(raster_meetnet_poly,
+           file.path(datapath, 
+                     "GIS/VoorR/raster_meetnet_poly.gpkg"), 
+           layer = "raster_meetnet_poly", 
+           driver = "GPKG",
+           delete_dsn = TRUE)
+  
+  
+  #overlay maken hab-kaart en GRTS-rooster
+  habmap_gw_raster_overlay <- habmap_polygons_gw %>% 
+    st_intersection(raster_meetnet_poly)
+  
+  habmap_gw_raster_overlay <- habmap_gw_raster_overlay %>% 
+      mutate(opp = as.integer(st_area(habmap_gw_raster_overlay))) 
+  
+  #wegschrijven van deze intersectie
+  st_write(habmap_gw_raster_overlay,
+           file.path(datapath, 
+                     "GIS/VoorR/habmap_gw_raster_overlay.gpkg"), 
+           layer = "habmap_gw_raster_overlay", 
+           driver = "GPKG",
+           delete_dsn = TRUE)
+  
+  
+  #inlezen grts-raster level 1 (hoogste resolutie = kleinste gridcelgrootte)
+  grts_level1 <- read_GRTSmh(brick = TRUE) %>% 
+    raster::subset(1)
```

Comment:
zonder al de hele context te kennen (mijn oog valt er even op); als je hier doelt op het fijnste raster (met elke 32 m cel een ander adres), dan heb je level 0 nodig, niet level 1 (64 m resolutie). Maar ik vermoed dat het een bewuste keuze is

Reply:
level 0 krijg je ook met plain `read_GRTSmh()`

Reply:
Mijn frank valt nu wat je doet. Er is geen probleem. Wel wat verwarring. In het resultaat van `read_GRTSmh(brick = TRUE)` zijn de brick layers gelabeld van `level0` tot `level9`. `subset(1)` is dus level0 (32m); je benoemt het als level1. Analoog voor level8 (resol 8192 m) = `subset(9)`

Je kunt gewoon als volgt werken: `... <- read_GRTSmh(brick = TRUE)$level0`

Reply:
de levelnummers hebben te maken met de exponent van factor 2 in `32 * 2^(0:9)`

Reply:
Ik heb de naamgeving logischer gemaakt level1 -> level0; level9 -> level8